src/scriptworker/cot/verify.py

```diff
@@ -1303,7 +1303,9 @@ async def get_in_tree_template(link):
         raise CoTError("{} source url {} doesn't end in .yml or .yaml!".format(link.name, source_url))
 
     auth = None
-    if (any(vcs_rule.get("require_secret") for vcs_rule in context.config["trusted_vcs_rules"])) and "github.com" in source_url:
+    if (
+        source_url.startswith("ssh://") or any(vcs_rule.get("require_secret") for vcs_rule in context.config["trusted_vcs_rules"])
+    ) and "github.com" in source_url:
```

Comment:
I'd be ok if you wanted to drop `require_secret` entirely -- guardian no longer uses it... but its fine to leave.

Reply:
Hm. XPI doesn't have the requirement of releasing off these private source repos (we release off the manifest repo, which is public and known). We still have a check in `is_pull_request` that checks for `has_commit_landed_on_repository`. That check is essentially skipped if we have `require_secret` set. I don't know if we still need that behavior. I'll leave it for now, but we may want to revisit.

Reply:
we can't test `has_commit_landed_on_repository` for a private repository. I don't have a need for this skippage anymore for guardian at all, but its worth knowing that there is no other way to get this information [that I could find] which would let us use a token.

It's essentially github-web-secret-api

Reply:
https://github.com/mozilla-releng/scriptworker/issues/456 for followup.
I may need actual private repo releases for adhoc, so hopefully I'll have a better understanding in a bit.